### PR TITLE
*: merge waiting list and suspect regions

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: ref #8378

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/schedule/checker/checker_controller.go [hunk 1]
```diff
@@ -152,19 +150,25 @@ func (c *Controller) GetRuleChecker() *RuleChecker {
 	return c.ruleChecker
 }
 
-// GetWaitingRegions returns the regions in the waiting list.
-func (c *Controller) GetWaitingRegions() []*cache.Item {
-	return c.regionWaitingList.Elems()
+// GetPendingProcessedRegions returns the pending processed regions in the cache.
+func (c *Controller) GetPendingProcessedRegions() []uint64 {
+	pendingRegions := make([]uint64, 0)
```
Comment (lhy1024) [inline pkg/schedule/checker/checker_controller.go:155]: why copy it?

File: pkg/schedule/coordinator.go [hunk 2]
```diff
@@ -222,22 +214,15 @@ func (c *Coordinator) checkRegions(startKey []byte) (key []byte, regions []*core
 	return
 }
 
-func (c *Coordinator) checkSuspectRegions() {
-	for _, id := range c.checkers.GetSuspectRegions() {
+func (c *Coordinator) checkPendingProcessedRegions() {
+	ids := c.checkers.GetPendingProcessedRegions()
+	waitingListGauge.Set(float64(len(ids)))
```
Comment (lhy1024) [inline pkg/schedule/coordinator.go:219]: Does it need to be renamed?

File: pkg/schedule/checker/checker_controller.go (hunk 1 above)
Comment (rleungx) [reply pkg/schedule/checker/checker_controller.go:155]: `Elems()` returns `[]*Item`

**Review by lhy1024 — APPROVED** [review]

File: server/cluster/scheduling_controller.go [hunk 3]
```diff
@@ -403,25 +403,25 @@ func (sc *schedulingController) PauseOrResumeChecker(name string, t int64) error
...
-func (sc *schedulingController) GetSuspectRegions() []uint64 {
+// GetPendingProcessedRegions gets all suspect regions.
+func (sc *schedulingController) GetPendingProcessedRegions() []uint64 {
 	sc.mu.RLock()
 	defer sc.mu.RUnlock()
-	return sc.coordinator.GetCheckerController().GetSuspectRegions()
+	return sc.coordinator.GetCheckerController().GetPendingProcessedRegions()
 }
 
 // RemoveSuspectRegion removes region from suspect list.
```
Comment (HuSharp) [inline server/cluster/scheduling_controller.go:420]:
```suggestion
// RemovePendingProcessedRegion removes pending processed region.
```

**Review by HuSharp — COMMENTED** [review]
rest LGTM

**Review by HuSharp — APPROVED** [review]

**Pull request merged**